webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/index.js

```diff
@@ -0,0 +1,29 @@
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as actions from './TypeAheadSelectActions';
+import reducer from './TypeAheadSelectReducer';
+import TypeAheadSelect from './TypeAheadSelect';
+
+const selectTypeAheadSelect = ({ typeAheadSelect }, id) => typeAheadSelect[id];
+
+const selectOptions = (selected, ownProps) =>
+  selected ? selected.options : ownProps.options;
+
+const selectSelected = (selected, ownProps) =>
+  selected ? selected.selected : ownProps.selected;
+
+const mapStateToProps = (state, ownProps) => {
+  const { id } = ownProps;
+  const selected = selectTypeAheadSelect(state, id);
+
+  return {
+    options: selectOptions(selected, ownProps),
+    selected: selectSelected(selected, ownProps),
+  };
+};
+
+const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);
+
+export const reducers = { typeAheadSelect: reducer };
+
+export default connect(mapStateToProps, mapDispatchToProps)(TypeAheadSelect);
```

Comment:
since we started to use hooks and `react-redux` supports it, we started to refactor components to a new convention
which doesn't use `mapStateToProps`, `connect`, `mapDispatchToProps`, could you change this file into something like:
```js
import React from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import TypeAheadSelect from './TypeAheadSelect';
import { selectState, selectTitle } from './PowerStatusSelectors';

const ConnectedTypeAheadSelect = ({ data: { id } }) => {
  const selected = useSelector(state => selectSelected(state, id));
  const options = useSelector(state => selectOptions(state, id));
  const dispatch = useDispatch();
  const onChange = items => dispatch(....)

  return <TypeAheadSelect selected={selected} options={options} onChange={onChange}/>;
};

ConnectedTypeAheadSelect.propTypes = {
  ...
};

export default ConnectedTypeAheadSelect;
```